src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt

```diff
@@ -73,21 +71,15 @@ data class RustToolchain(val location: String) {
         if (procOut.exitCode != 0 || procOut.isCancelled || procOut.isTimeout)
             return null
 
-        return extractor(procOut.stdoutLines)
+        return procOut.stdoutLines
     }
 
-    fun cargo(cargoProjectDirectory: String): Cargo =
-        Cargo(pathToExecutable(CARGO), pathToExecutable(RUSTC), cargoProjectDirectory)
-
-    fun nonProjectCargo(): Cargo =
-        Cargo(pathToExecutable(CARGO), pathToExecutable(RUSTC), null)
-
-    val presentableLocation: String = PathUtil.toPresentableUrl(pathToExecutable(CARGO))
-
-    private fun pathToExecutable(toolName: String): String {
-        val exeName = if (SystemInfo.isWindows) "$toolName.exe" else toolName
-        return File(File(location), exeName).absolutePath
-    }
+    data class VersionInfo(
```

Comment:
hope we can use [typealias](https://github.com/Kotlin/KEEP/issues/4) as quickly as possible so that it is only to create something like

```
typealias LabelVersion = (JLabel, Version?)

data class VersionInfo(
  val cargo: LabelVersion,
 ...
)
```

so that we can encode labels with relative version info directory with explicitly create a map to iterate over it.